x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/boxplot/BoxplotAggregatorFactory.java

```diff
@@ -24,6 +29,24 @@
 
     private final double compression;
 
+    static void registerAggregators(ValuesSourceRegistry valuesSourceRegistry) {
+        valuesSourceRegistry.register(BoxplotAggregationBuilder.NAME,
+            CoreValuesSourceType.NUMERIC, new BoxplotAggregatorSupplier() {
+                @Override
```

Comment:
@talevy showed me that in cases like this, we can use a lambda with a type cast.  Since `BoxplotAggregator` already just takes a `ValuesSource` directly (as opposed to needing a cast), I think we can just say `(BoxplotAggregationSupplier) BoxplotAggregator::new` here.  We'll also need to add the `@FunctionalInterface` annotation to `BoxplotAggregationSupplier`.

Reply:
:+1: